lua/sql/defs.lua

```diff
@@ -529,6 +607,80 @@ M.type_of_stmt_ptr = ffi.typeof "sqlite3_stmt*"
 M.type_of_exec_ptr = ffi.typeof "int (*)(void*,int,char**,char**)"
 M.type_of_blob_ptr = ffi.typeof "sqlite3_blob*"
 
+---Fail at lua side if sql table doesn't exists.
+---@param db table
+---@param tbl_name string
+---@param method any
+M.assert_tbl = function(db, tbl_name, method)
+  local errmsg = ("sql.nvim: can not execute %s, %s doesn't exists."):format(method, tbl_name)
+  assert(db:exists(tbl_name), errmsg)
+end
+
+--- Wrapper around clib.exec for convenience.
+---@param conn_ptr sqlite connction ptr
+---@param statement string: statement to be executed.
+---@return table: stmt object
+M.exec_stmt = function(conn_ptr, statement)
+  return clib.sqlite3_exec(conn_ptr, statement, nil, nil, nil)
+end
+
+--- Execute a manipulation sql statement within begin and commit block
+---@param conn_ptr sqlite connction ptr
+---@param sqldb table
+---@param fn func()
+M.wrap_stmts = function(conn_ptr, fn)
+  M.exec_stmt(conn_ptr, "BEGIN")
+  fn()
+  M.exec_stmt(conn_ptr, "COMMIT")
+  return
+end
+
+--- Wrapper around stmt module for convenience.
+---@param conn_ptr sqlite connction ptr
+---@param statement string: statement to be parsed.
+---@return table: stmt object
+M.parse_stmt = function(conn_ptr, statement)
+  return require("sql.stmt"):parse(conn_ptr, statement)
+end
```

Comment:
same. and you already know it by doing that to fix the recursive inclusion.

Reply:
what do you mean?

Reply:
stmt requires def requires stmt ?!?!

Reply:
defs should be the most low level layer and not require any abstraction

Reply:
Oh I see what you mean, coming to think about I don't even now why this endup here 😆 .